### PR TITLE
[DOCS-6114] Sync Service 3.4 docs

### PR DESCRIPTION
**Pull request by anxumalo**
Summary of changes:

- Copy previous version (Sync Service 3.3) from `latest` to `3.3` and change all links in this new folder to use `3.3`.
- Prepare Sync Service 3.4 docs in `latest` and change artefacts to `3.4.x`.
- Update the site configuration file (`_config.yml`) to define the releases required for the version drop-down selector in content pages.
- Update the table of contents (`_data/toc/sync-service.yaml`) to add the content structure for v3.3.

⚠️ This PR is set to WIP until all sub-tasks (#250) have been reviewed, approved, and merged into the release branch.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -739,15 +739,20 @@ defaults:
...
         - 3.1
         - 3.0
   - scope:
       path: "sync-service/latest"
     values:
-      version: 3.3
+      version: 3.4
       latest: true
+  - scope:
+      path: "sync-service/3.3"
```
Comment (ciprianchelaru) [inline _config.yml:753]: why path: "sync-service/3.3" shouldn't be "sync-service/3.4" ?

File: _config.yml [hunk 2]
```diff
@@ -739,15 +739,20 @@ defaults:
...
+    values:
+      version: 3.3
```
Comment (ciprianchelaru) [inline _config.yml:755]: the same question as above

File: _data/toc/sync-service.yaml [hunk 3]
```diff
@@ -37,6 +37,48 @@
           path: '/sync-service/latest/admin/troubleshoot/'
         - title: 'FAQ'
           path: '/sync-service/latest/admin/faq/'
+
+# Alfresco Sync Service 3.3
```
Comment (ciprianchelaru) [inline _data/toc/sync-service.yaml:41]: shouldn't make a reference for 3.4 ?
please see also the same below references

**Review by ciprianchelaru — COMMENTED** [review]
please see my questions

File: _config.yml (hunk 1 above)
Comment (anxumalo) [reply _config.yml:753]: Great questions! This config file is used to build the Docs site, and sets the versions shown in the content pages. I've added an explanation of what's going on in the PR description. This should address all your questions so far.

So, in this file, the key changes are:
- moving v3.3 content from "latest" to "sync-service/3.3"
- setting v3.4 as the "latest" version (see lines 746 - 750)

HTH

File: _config.yml (hunk 2 above)
Comment (anxumalo) [reply _config.yml:755]: See earlier comments

File: _data/toc/sync-service.yaml (hunk 3 above)
Comment (anxumalo) [reply _data/toc/sync-service.yaml:41]: See earlier comments

**Review by ciprianchelaru — APPROVED** [review]
Thanks for clarifications! :)

**Review by oliveralfresco2018 — APPROVED** [review]
lgtm

**Pull request merged**